internal/aws/awsutil/conn.go

```diff
@@ -203,29 +203,58 @@ func GetAWSConfigSession(logger *zap.Logger, cn ConnAttr, cfg *AWSSessionSetting
 		HTTPClient:                    http,
 		CredentialsChainVerboseErrors: aws.Bool(true),
 	}
-	// do not overwrite for sts assume role
-	if cfg.RoleARN == "" && len(override.GetCredentialsChainOverride().GetCredentialsChain()) > 0 {
-		config.Credentials = credentials.NewCredentials(&credentials.ChainProvider{
-			Providers: customCredentialProvider(cfg, config),
-		})
-	}
 	return config, s, nil
 }
 
+// order
+// 1. shared profile
+// 2. env var
+// 3. shared creds file
+// 4. ec2 creds
+// 5. override creds providers
+// explicit keys do not make sense in this context since contrib does not take in explicit keys
+// and cwa does not provide explicit keys to config but come from these methods of getting creds
+// @TODO make this more upstream friendly working in the future
 func customCredentialProvider(cfg *AWSSessionSettings, config *aws.Config) []credentials.Provider {
```

Comment:
To recap the discussion we had earlier, this should only be the custom credential chain that'll sit before the SDK credential chain. The override provider should be appended to the front of the chain. I still think we can follow the agent and have the roleARN STS provider be an alternative chain to the refreshable shared credential providers. https://github.com/aws/amazon-cloudwatch-agent/blob/main/cfg/aws/credentials.go#L123